.github/workflows/lint.yml

```diff
@@ -2,33 +2,27 @@
 
 name: lint
 
-# Controls when the action will run.
 on:
     push:
         branches:
           - main
     pull_request:
 
-
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   lint:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
         with:
-          python-version: "3.8"
-
+          python-version: "3.9"
```

Comment:
The docs state "SCICO requires Python version 3.7 or later." Given that, I suggest that it would be safer to run the tests on 3.7, as I assume code that works in 3.7 will also work in 3.9 but probably not the reverse.

Reply:
I'm not sure I agree. I would argue that it's more important to test against the "recommended" version of Python than the minimal supported version. It's perhaps a bit more likely that code that works in 3.7 will work in 3.9 than vice-versa, but it's far more serious if it doesn't work in 3.9 since 3.7 is only supported to allow use of google colab. Perhaps a reasonable alternative would be to introduce a periodic (weekly or montly) test across multiple Python version, along the same lines as the existing "jax latest" test.